python_modules/dagster/dagster/_core/execution/context/output.py

```diff
@@ -325,7 +325,7 @@ def asset_key(self) -> AssetKey:
 
     @public
     @property
-    def asset_partitions_def(self) -> "PartitionsDefinition":
+    def partitions_def(self) -> "PartitionsDefinition":
```

Comment:
Concerned that this method will not work for op job execution -- [op jobs can also execute with partitions, where each op has the same partitions def](https://docs.dagster.io/concepts/partitions-schedules-sensors/partitioning-ops).

Since we're moving to a more general naming, I think we need to support both the asset and op use cases within each context method.

So for the `OutputContext.partitions_def` method, that would involve:
- First checking if the output is an asset output
- If so, using the existing logic in this method to return the partitions def from the asset layer
- If not, returning the partitions def from the job def (think we might be able to call `step_context.partitions_def`)